packages/material-ui/src/SwipeableDrawer/SwipeableDrawer.js

```diff
@@ -284,29 +294,47 @@ class SwipeableDrawer extends React.Component {
           ? window.innerHeight - event.changedTouches[0].clientY
           : event.changedTouches[0].clientY;
     }
-    const translateRatio = this.getTranslate(current) / this.getMaxTranslate();
 
-    // We have to open or close after setting swiping to null,
-    // because only then CSS transition is enabled.
-    if (translateRatio > 0.5) {
-      if (this.isSwiping && !this.props.open) {
-        // Reset the position, the swipe was aborted.
-        this.setPosition(this.getMaxTranslate(), {
-          mode: 'enter',
+    // get the beginning of the swipe, pending on orientation
+    const beginSwipe = isHorizontal(this.props) ? this.startX : this.startY;
+
+    // compute minimal swipe distance by hysteresis property
+    const minDistance = this.getMaxTranslate() * this.props.hysteresis;
+
+    // get the absolute distance
+    const distance = Math.abs(beginSwipe - current);
+
+    // get elapsed time by compute difference from point of time of exec. 
+    // handleBodyTouchStart() to now
+    const elapsedTime = performance.now() - this.touchStartTime;
+
+    // velocity in px/ms
+    const velocity = distance / elapsedTime;
```

Comment:
What do you think of using a low pass filter to stabilize the computed value? Also, assuming the mousemove events are trigger more or less after a fixed period, we can remove the elapse time computation. I was able to condense the implementation down to this: https://github.com/oliviertassinari/react-swipeable-views/blob/01f419c8246ca6ddeed1178e54d4c6eec670baa7/packages/react-swipeable-views/src/SwipeableViews.js#L480

Reply:
> Also, assuming the mousemove events are trigger more or less after a fixed period, we can remove the elapse time computation

Calculating a velocity without taking the time into account seems like an ugly hack to me.